localstack/utils/docker_utils.py

```diff
@@ -278,6 +278,31 @@ def __iter__(self):
         return self.mappings.__iter__()
 
 
+@dataclasses.dataclass
+class ContainerConfiguration:
```

Comment:
are there any plans to merge this with the LocalstackContainerConfiguration?

Reply:
Yes, that was the plan. However, I did not do it for this PR since, since a lot of things (except the image) are optional in the ContainerConfiguration, and therefore cannot be non-optional in the LocalStackContainerConfiguration, so I need to refactor that a bit.